Scripts/RTS/Chunk.cs

```diff
@@ -1,32 +1,47 @@
 ﻿namespace RTS;
+using System.Collections.Generic;
 
 public class Chunk
 {
     TileMap tileMap;
 
-    public Chunk(int chunkX, int chunkY, TileMap tileMap, Noise noise)
+    public Chunk(int chunkX, int chunkY, TileMap tileMap, Noise noise, bool generate = true)
     {
         this.tileMap = tileMap;
+        if(generate) this.GenerateWorld(chunkX, chunkY, tileMap, noise);
+    }
 
+    void GenerateWorld(int chunkX, int chunkY, TileMap tileMap, Noise noise)
+    {
         for (int x = 0; x < World.ChunkSize; x++)
         {
             for (int y = 0; y < World.ChunkSize; y++)
             {
                 var globalX = (chunkX * World.ChunkSize) + x;
                 var globalY = (chunkY * World.ChunkSize) + y;
 
-                string type;
+                string type = "";
+                var currentNoise = noise.GetNoise2D(globalX, globalY);
+                var name = ((string) (this.tileMap.Name)).ToLower();
```

Comment:
.ToString() looks better imo

Reply:
Why would you get the name of the tile from the tileMap node name? Tiles are within the tilemap in the tileset. This does not make sense to me.